src/components/PostDetails.tsx

```diff
@@ -1,61 +1,45 @@
 import React, { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
 
 import * as commentsApi from '../api/comments';
 
 import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
+import { CommentData } from '../types/Comment';
+import { useAppSelector } from '../app/hooks';
+import {
+  setLoaded,
+  setHasError,
+  setComments,
+} from '../features/comments/commentsSlice';
 
 type Props = {
   post: Post;
 };
 
 export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
   const [visible, setVisible] = useState(false);
+  const dispatch = useDispatch();
+  const {
+    loaded,
+    hasError,
+    comments,
+  } = useAppSelector(state => state.comments);
 
   function loadComments() {
-    setLoaded(false);
-    setError(false);
+    dispatch(setLoaded(false));
+    dispatch(setHasError(false));
     setVisible(false);
 
     commentsApi.getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
+      .then(commentsFromServer => dispatch(setComments(commentsFromServer)))
+      .catch(() => dispatch(setHasError(true)))
+      .finally(() => dispatch(setLoaded(true)));
   }
 
   useEffect(loadComments, [post.id]);
```

Comment:
The dependency array for the useEffect should be updated to include the `loadComments` function. Change the dependency array to `[post.id, loadComments]`.